routing/datasource.go

```diff
@@ -140,10 +140,17 @@ func applyIncoming(defs routeDefs, d *incomingData) routeDefs {
 	return defs
 }
 
+type mergedDefs struct {
+	routes  []*eskip.Route
+	clients map[DataClient]struct{}
+}
+
 // merges the route definitions from multiple data clients by route id
-func mergeDefs(defsByClient map[DataClient]routeDefs) []*eskip.Route {
+func mergeDefs(defsByClient map[DataClient]routeDefs) mergedDefs {
+	clients := make(map[DataClient]struct{}, len(defsByClient))
 	mergeByID := make(routeDefs)
-	for _, defs := range defsByClient {
+	for c, defs := range defsByClient {
+		clients[c] = struct{}{}
```

Comment:
This assumes no client is repeated (which actually how skipper works now), but can override if we use multiple same dataclient multiple times

Reply:
I've created follow up #3527